### PR TITLE
add basic support for phone numbers api

### PR DESCRIPTION
**Pull request by MiTo0o**
- generate random phone number based on country code WIP (LONG list of countries)
- generate random phone number with random format
- generate random IMEI number

### CODE REVIEW

**Comment by MiTo0o** [comment]
was wondering if this looks ok? Phone numbers can also be formatted with dashes like `111-111-111` or with () like `(111)-111-111`. I currently just left it with spaces like `+1 111 111 111`.

**Comment by MiTo0o** [comment]
now that I think about it, my whole approach may have been overkill. Most people probably don't care that much about a specific country's phone number format.

**Comment by MiTo0o** [comment]
ah, and most countries have many different phone formats locally. I'm just gonna wait for some suggestions.

**Comment by ageddesi** [comment]
Hi @MiTo0o,
Thank you for the PR, this is great work. I love that you added the ability to have country codes. 
In regards to your question about spaces. Can we make it an option in the url. so you can choose the format.

Examples are:
spaces
no-spaces
dashes

What do you think?

**Comment by MiTo0o** [comment]
> Hi @MiTo0o, Thank you for the PR, this is great work. I love that you added the ability to have country codes. In regards to your question about spaces. Can we make it an option in the url. so you can choose the format.
> 
> Examples are: spaces no-spaces dashes
> 
> What do you think?

Sounds great, I will add those three options.

**Comment by MiTo0o** [comment]
hey @ageddesi do you think it's ok if we only provide the phone number formats to popular countries for now? It's a lot of work to search every countries phone number format. Some countries also have many different phone number formats.

**Comment by ageddesi** [comment]
@MiTo0o Hi, do how ever many you like for this step and then please can you raise a ticket for what needs to be done. Hopefully someone else will pick it up as part of Hacktoberfest

File: modules/phone-numbers/api/phone-numbers-routes.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import e, { Request, Response } from 'express';
+import * as core from 'express-serve-static-core';
+import { faker } from "@faker-js/faker";
+import countryNumberData from '../consts/countryNumberData';
+import supportedCountries from '../utils/supportedCountries';
+
+const formats = {
```
Comment (ageddesi) [inline modules/phone-numbers/api/phone-numbers-routes.ts:7]: Do you mind making this a const in the consts folder so it can be used in the swagger docs.

**Comment by MiTo0o** [comment]
@ageddesi Thank you for your patience and quick responses :D. How does it look so far? Is there anything else I should clean up / do. I think I'll take your suggestion and raise a ticket for the rest of the phone number formats.

**Comment by MiTo0o** [comment]
hey @ageddesi I merged the 0.13.0 branch into mine and found some problems with swagger.json, there were missing/extra commas and there are some duplicate keys. I fixed the commas, but not sure what you want to do with the duplicate keys.

**Comment by ageddesi** [comment]
Hi @MiTo0o, I have just pulled in a new way of doing the swagger docs. 
We now do the openAPI spec directly in the API file itself using comments. 
If you update your code to use this you can ignore the swagger.json file as it will auto generate.

**Comment by MiTo0o** [comment]
oh the openAPI seems so cool, let me check it out and update

**Comment by ageddesi** [comment]
Thanks @MiTo0o sorry to be a pain, someone kindly finally set it up for me as part of hacktoberfest and handles all the conflicts I have been having

**Comment by MiTo0o** [comment]
hey @ageddesi its all good :D. I'm always open to learning more. One quick thing, how do I generate swagger with openAPI? and also getting a error in another api route

image
image

the Music interface is being imported by also being created again the the same file

**Comment by ageddesi** [comment]
I pushed a fix for this a while ago on the v.0.13.0 branch. If not just
delete the inline model.

**Comment by ageddesi** [comment]
@MiTo0o Once you have added the OpenAPI comments to your api ts file. The next time you run the application and go to the localhost url you will see your changes refelected in the docs.

**Comment by MiTo0o** [comment]
ok so the thing is my fork only has the main branch, I can't merge 0.13.0 branch into my branch ;(

**Comment by ageddesi** [comment]
ok no problem, leave it with me 👍 I will get what you have done merged in.

**Comment by ageddesi** [comment]
@all-contributors please add @MiTo0o for code. 

Thanks for the contribution, Happy Hacktoberfest. I hope you will want to do more in our API. If you do like what we are doing please consider staring the repo and joining our Discord channel at https://discord.gg/yxqrkjDR87

**Comment by MiTo0o** [comment]
> Thanks for the contribution, Happy Hacktoberfest. I hope you will want to do more in our API. If you do like what we are doing please consider staring the repo and joining our Discord channel at https://discord.gg/yxqrkjDR87

Thank you so much! :D I had a lot of fun

**Pull request merged**